Waveshare epaper not working after update to 1.13.0

<!-- Thanks for reporting a bug for this project. READ THIS FIRST:

- Provide as many details as possible. Simply saying "X gives bug" or "X gives error" is not enough!
- Paste logs, configuration sample and code into the backticks (```).
- Read through the template carefully and fill out all missing details.
- Please also search for similar issues in this issue tracker first and read through the ESPHome FAQ.

DO NOT DELETE ANY TEXT from this template! Otherwise the issue may be closed without a comment.
-->

**Operating environment/Installation (Hass.io/Docker/pip/etc.):**
<!--
Please provide details about your environment below this line. -->
Hass.io


**ESP (ESP32/ESP8266, Board/Sonoff):**
<!--
Please provide details about which ESP you're using below.
-->
Nodemcu v3

**Affected component:**
<!--
Please add the link to the documentation at https://esphome.io/index.html of the component in question.
-->
https://esphome.io/components/display/waveshare_epaper.html

**Description of problem:**
I started experimenting with Waveshare's epaper panel the other day and wrote some basic code to test. Worked perfectly fine with 1.12.2 as far as compiling and uploading goes (still have it loaded on the nodemcu). 
But after today's upgrade to 1.13.0, I can no longer compile my code. I'm still learning esphome and am not a professional software dev, so my powers of troubleshooting are still limited. 

**Problem-relevant YAML-configuration entries:**
```yaml
esphome:
  name: wall_clock
  platform: ESP8266
  board: nodemcuv2

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

# Enable logging
logger:

# Enable Web server
web_server:
  port: 80

# Enable Home Assistant API
api:
  password: !secret ESPHomeAPI

ota:
  password: !secret ESPHomeOTA

time:
  - platform: homeassistant
    id: homeassistant_time
    
font:
  - file: "fonts/Roboto-Regular.ttf"
    id: roboto_regular_20
    size: 20
  
# Waveshare epaper config
spi:
  clk_pin: D0
  mosi_pin: D1

display:
  platform: waveshare_epaper
  cs_pin: D2
  dc_pin: D3
  busy_pin: D4
  reset_pin: D5
  model: 7.50in
  lambda: |-
    it.print(0, 0, id(roboto_regular_20), "Text test");    
```

**Traceback (if applicable):**
<!--
Please copy the traceback here if compilation is failing. If possible, also connect to the ESP over USB and copy its logs into the backticks.
-->
```
INFO Reading configuration...
INFO Detected timezone 'CET' with UTC offset 1 and daylight savings time from 03/29/20 02:00:00 to 10/25/20 03:00:00
INFO Generating C++ source...
INFO Compiling app...
INFO Running:  platformio run -d /config/esphome/wall_clock
Processing wall_clock (framework: arduino; platform: espressif8266@1.8.0; board: nodemcuv2)
--------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif8266/nodemcuv2.html
PLATFORM: Espressif 8266 > NodeMCU 1.0 (ESP-12E Module)
HARDWARE: ESP8266 80MHz 80KB RAM (4MB Flash)
Library Dependency Finder -> http://bit.ly/configure-pio-ldf
LDF MODES: FINDER(chain) COMPATIBILITY(soft)
Collected 29 compatible libraries
Scanning dependencies...
Dependency Graph
|-- <ArduinoJson-esphomelib> 5.13.3
|-- <ESP Async WebServer> 1.1.1
|   |-- <ESPAsyncTCP> 1.2.0
|   |   |-- <ESP8266WiFi> 1.0
|   |-- <Hash> 1.0
|   |-- <ESP8266WiFi> 1.0
|   |-- <ArduinoJson-esphomelib> 5.13.3
|-- <ESP8266WiFi> 1.0
|-- <ESP8266mDNS>
|   |-- <ESP8266WiFi> 1.0
|-- <ESPAsyncTCP> 1.2.0
|   |-- <ESP8266WiFi> 1.0
Compiling /data/wall_clock/.pioenvs/wall_clock/src/main.cpp.o
src/main.cpp: In function 'void setup()':
src/main.cpp:224:42: error: expected type-specifier before 'waveshare_epaper'
waveshare_epaper_waveshareepaper = new waveshare_epaper();
^
src/main.cpp:224:42: error: expected ';' before 'waveshare_epaper'
src/main.cpp:225:37: error: request for member 'set_update_interval' in '* waveshare_epaper_waveshareepaper', which is of non-class type 'esphome::waveshare_epaper::WaveshareEPaperTypeBModel'
waveshare_epaper_waveshareepaper->set_update_interval(1000);
^
src/main.cpp:228:37: error: request for member 'set_update_interval' in '* waveshare_epaper_waveshareepaper', which is of non-class type 'esphome::waveshare_epaper::WaveshareEPaperTypeBModel'
waveshare_epaper_waveshareepaper->set_update_interval(1000);
^
src/main.cpp:231:37: error: request for member 'set_spi_parent' in '* waveshare_epaper_waveshareepaper', which is of non-class type 'esphome::waveshare_epaper::WaveshareEPaperTypeBModel'
waveshare_epaper_waveshareepaper->set_spi_parent(spi_spicomponent);
^
src/main.cpp:232:37: error: request for member 'set_cs_pin' in '* waveshare_epaper_waveshareepaper', which is of non-class type 'esphome::waveshare_epaper::WaveshareEPaperTypeBModel'
waveshare_epaper_waveshareepaper->set_cs_pin(new GPIOPin(4, OUTPUT, false));
^
src/main.cpp:233:37: error: request for member 'set_dc_pin' in '* waveshare_epaper_waveshareepaper', which is of non-class type 'esphome::waveshare_epaper::WaveshareEPaperTypeBModel'
waveshare_epaper_waveshareepaper->set_dc_pin(new GPIOPin(0, OUTPUT, false));
^
src/main.cpp:234:37: error: request for member 'set_writer' in '* waveshare_epaper_waveshareepaper', which is of non-class type 'esphome::waveshare_epaper::WaveshareEPaperTypeBModel'
waveshare_epaper_waveshareepaper->set_writer([=](display::DisplayBuffer & it) -> void {
^
src/main.cpp:237:37: error: request for member 'set_reset_pin' in '* waveshare_epaper_waveshareepaper', which is of non-class type 'esphome::waveshare_epaper::WaveshareEPaperTypeBModel'
waveshare_epaper_waveshareepaper->set_reset_pin(new GPIOPin(14, OUTPUT, false));
^
src/main.cpp:238:37: error: request for member 'set_busy_pin' in '* waveshare_epaper_waveshareepaper', which is of non-class type 'esphome::waveshare_epaper::WaveshareEPaperTypeBModel'
waveshare_epaper_waveshareepaper->set_busy_pin(new GPIOPin(2, INPUT, false));
^
In file included from src/esphome.h:23:0,
from src/main.cpp:3:
src/esphome/core/application.h: In instantiation of 'C* esphome::Application::register_component(C*) [with C = esphome::waveshare_epaper::WaveshareEPaperTypeBModel]':
src/main.cpp:226:58:   required from here
src/esphome/core/application.h:81:5: error: static assertion failed: Only Component subclasses can be registered
static_assert(std::is_base_of<Component, C>::value, "Only Component subclasses can be registered");
^
*** [/data/wall_clock/.pioenvs/wall_clock/src/main.cpp.o] Error 1
========================== [ERROR] Took 2.63 seconds ==========================
```

**Additional information and things you've tried:**
I tried: 
- clean build
- delete device and recreate from scratch

Nothing helped. 


<!-- LEAVE THIS LINE AS-IS AND DON'T DELETE IT, OTHERWISE THE ISSUE WILL BE CLOSED AUTOMATICALLY. -->

--- a/esphome/components/waveshare_epaper/display.py
+++ b/esphome/components/waveshare_epaper/display.py
@@ -24,7 +24,7 @@ MODELS = {
     '2.90in': ('a', WaveshareEPaperTypeAModel.WAVESHARE_EPAPER_2_9_IN),
     '2.70in': ('b', WaveshareEPaper2P7In),
     '4.20in': ('b', WaveshareEPaper4P2In),
-    '7.50in': ('b', WaveshareEPaperTypeBModel),
+    '7.50in': ('b', WaveshareEPaper7P5In),
 }
 
 

--- a/esphome/platformio_api.py
+++ b/esphome/platformio_api.py
@@ -7,6 +7,7 @@ import re
 import subprocess
 
 from esphome.core import CORE
+from esphome.py_compat import IS_PY2
 from esphome.util import run_external_command, run_external_process
 
 _LOGGER = logging.getLogger(__name__)
@@ -17,12 +18,10 @@ def patch_structhash():
     # removed/added. This might have unintended consequences, but this improves compile
     # times greatly when adding/removing components and a simple clean build solves
     # all issues
+    # pylint: disable=no-member,no-name-in-module
     from platformio.commands import run
     from platformio import util
-    try:
-        from platformio.util import get_project_dir
-    except ImportError:
-        from platformio.project.helpers import get_project_dir
+    from platformio.util import get_project_dir
     from os.path import join, isdir, getmtime, isfile
     from os import makedirs
 
@@ -69,7 +68,8 @@ def run_platformio_cli(*args, **kwargs):
     if os.environ.get('ESPHOME_USE_SUBPROCESS') is None:
         import platformio.__main__
         try:
-            patch_structhash()
+            if IS_PY2:
+                patch_structhash()
         except Exception:  # pylint: disable=broad-except
             # Ignore when patch fails
             pass

